src/ott/neural/flows/flows.py

```diff
@@ -0,0 +1,157 @@
+
+    Args:
+      t: Time :math:`t`.
+    """
+
+  @abc.abstractmethod
+  def compute_ut(
+      self, t: jnp.ndarray, src: jnp.ndarray, tgt: jnp.ndarray
+  ) -> jnp.ndarray:
+    """Evaluate the conditional vector field.
+
+    Evaluate the conditional vector field defined between :math:`x_0` and
+    :math:`x_1` at time :math:`t`.
+
+    Args:
+      t: Time :math:`t`.
+      src: Sample from the source distribution.
+      tgt: Sample from the target distribution.
+    """
+
+  def compute_xt(
+      self, noise: jnp.ndarray, t: jnp.ndarray, src: jnp.ndarray,
+      tgt: jnp.ndarray
+  ) -> jnp.ndarray:
+    """Sample from the probability path.
+
+    Sample from the probability path between :math:`x_0` and :math:`x_1` at
+    time :math:`t`.
+
+    Args:
+      noise: Noise sampled from a standard normal distribution.
+      t: Time :math:`t`.
+      src: Sample from the source distribution.
+      tgt: Sample from the target distribution.
+
+    Returns:
+      Samples from the probability path between :math:`x_0` and :math:`x_1`
+      at time :math:`t`.
+    """
+    mu_t = self.compute_mu_t(t, src, tgt)
+    sigma_t = self.compute_sigma_t(t)
+    return mu_t + sigma_t * noise
+
+
+class StraightFlow(BaseFlow, abc.ABC):
+  """Base class for flows with straight paths."""
+
+  def compute_mu_t(  # noqa: D102
+      self, t: jnp.ndarray, src: jnp.ndarray, tgt: jnp.ndarray
+  ) -> jnp.ndarray:
+    return t * src + (1 - t) * tgt
+
+  def compute_ut(
+      self, t: jnp.ndarray, src: jnp.ndarray, tgt: jnp.ndarray
+  ) -> jnp.ndarray:
+    """Evaluate the conditional vector field.
+
+    Evaluate the conditional vector field defined between :math:`x_0` and
+    :math:`x_1` at time :math:`t`.
+
+    Args:
+      t: Time :math:`t`.
+      src: Sample from the source distribution.
+      tgt: Sample from the target distribution.
+
+    Returns:
+      Conditional vector field evaluated at time :math:`t`.
+    """
+    return tgt - src
+
+
+class ConstantNoiseFlow(StraightFlow):
+  r"""Flow with straight paths and constant flow noise :math:`\sigma`."""
+
+  def compute_sigma_t(self, t: jnp.ndarray) -> jnp.ndarray:
+    r"""Compute noise of the flow at time :math:`t`.
+
+    Args:
+      t: Time :math:`t`.
+
+    Returns:
+      Constant, time-independent standard deviation :math:`\sigma`.
+    """
+    return self.sigma
+
+
+class BrownianNoiseFlow(StraightFlow):
+  r"""Brownian Bridge Flow.
+
+  Sampler for sampling noise implicitly defined by a Schroedinger Bridge
+  problem with parameter `\sigma` such that
+  :math:`\sigma_t = \sigma * \sqrt(t * (1-t))`.
+  """
+
+  def compute_sigma_t(self, t: jnp.ndarray) -> jnp.ndarray:
+    """Compute the standard deviation of the probablity path at time :math:`t`.
+
+    Args:
+      t: Time :math:`t`.
+
+    Returns:
+      Standard deviation of the probablity path at time :math:`t`.
+    """
+    return jnp.sqrt(self.sigma * t * (1 - t))
```

Comment:
The sigma should be outside the sqrt: 
```suggestion
    return self.sigma * jnp.sqrt(t * (1 - t))
```